File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        hashed = {}

        for i in range(len(nums)):
            x = target - nums[i]
            if x in hashed:
                return [i, hashed[x]]
            hashed[nums[i]] = i
        return []

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashed;
        for (int i = 0 ; i < nums.size() ; i++){
            int x = target - nums[i];
            if (hashed.count(x)){
                return {i, hashed[x]};
            }
                
            hashed[nums[i]] = i;
        }
        return {};
    }
};